src/Tests/SFA.DAS.ApprenticeFeedback.Api.UnitTests/Controllers/FeedbackTransactions/WhenTrackEmailTransactionClick.cs

```diff
@@ -14,44 +15,55 @@
 
 namespace SFA.DAS.ApprenticeFeedback.Api.UnitTests.Controllers.FeedbackTransaction
 {
-    public class TrackEmailTransactionClick
+    public class WhenTrackEmailTransactionClick
     {
         FeedbackTransactionController _sut;
-        Mock<IMediator> _mockMediator;
+        Mock<IMediator> _mediator;
+        Mock<IBackgroundTaskQueue> _backgroundTaskQueue;
 
         [SetUp]
         public void Setup()
         {
-            _mockMediator = new Mock<IMediator>();
-            _sut = new FeedbackTransactionController(_mockMediator.Object, Mock.Of<ILogger<FeedbackTransactionController>>());
+            _mediator = new Mock<IMediator>();
+            _backgroundTaskQueue = new Mock<IBackgroundTaskQueue>();
+            _sut = new FeedbackTransactionController(_mediator.Object, _backgroundTaskQueue.Object, Mock.Of<ILogger<FeedbackTransactionController>>());
         }
 
         [Test, RecursiveMoqAutoData]
         public async Task And_MediatorCommandSuccessful_Then_ReturnOk(
             TrackEmailTransactionClickCommand command,
             TrackEmailTransactionClickResponse response)
         {
-            _mockMediator.Setup(s => s.Send(
+            // Arrange
+            _mediator.Setup(s => s.Send(
                 It.IsAny<TrackEmailTransactionClickCommand>(),
                 It.IsAny<CancellationToken>())
             ).ReturnsAsync(response);
 
+            // Act
             var result = await _sut.TrackEmailTransactionClick(command.FeedbackTransactionId, command);
+            
+            // Assert
             result.Should().BeOfType<OkObjectResult>().Which.Value.Should().BeEquivalentTo(response);
         }
 
         [Test, MoqAutoData]
         public async Task And_MediatorThrowsException_Then_ReturnBadRequest(
             TrackEmailTransactionClickCommand command)
         {
-            _mockMediator.Setup(s => s.Send(
+            // Arrange
+            _mediator.Setup(s => s.Send(
                 It.IsAny<TrackEmailTransactionClickCommand>(),
                 It.IsAny<CancellationToken>())
             ).Throws(new Exception());
 
+            // Act
             var result = await _sut.TrackEmailTransactionClick(command.FeedbackTransactionId, command);
+            
+            // Assert
             result.Should().BeOfType<ObjectResult>();
```

Comment:
Should `AssertsonScope()` not be used here when assert multiple results?

Reply:
Removed the first assert instead as the cast to ObjectResult will cause a failure if it cannot be casted